domain/consensus/utils/utxo/utxo_iterator_with_diff.go

```diff
@@ -40,9 +40,19 @@ func IteratorWithDiff(iterator model.ReadOnlyUTXOSetIterator, diff model.UTXODif
 	}, nil
 }
 
-func (r *readOnlyUTXOIteratorWithDiff) First() {
-	r.baseIterator.First()
+func (r *readOnlyUTXOIteratorWithDiff) First() bool {
+	baseNotEmpty := r.baseIterator.First()
 	r.toAddIterator.First()
+
+	if !baseNotEmpty {
```

Comment:
What if base is empty but toAdd is not?

Also, I don't like the double-negation here, would rather if you did 
```
baseEmpty := !r.baseIterator.First()
if baseEmpty { ///... }
```